compiler/testData/codegen/box/annotations/annotatedLambda/samFunExpression.kt

```diff
@@ -35,15 +34,28 @@ fun testMethod(method: Method, name: String) {
     }
 }
 
-fun testClass(clazz: Class<*>, name: String) {
-    val invokes = clazz.getDeclaredMethods().single() { !it.isBridge() }
-    testMethod(invokes, name)
+fun testMethodNoAnnotations(method: Method, name: String) {
+    assertEquals(0, method.getDeclaredAnnotations().size, "No method annotations expected `$name`")
+    for ((index, annotations) in method.getParameterAnnotations().withIndex()) {
+        assertEquals(0, annotations.size, "No parameter $index annotations expected `$name`")
+    }
 }
 
-fun box(): String {
-    testClass(Test.apply(@Ann("OK") fun(){}), "1")
 
-    testClass(Test.applyABC(@Ann("OK") fun(@Ann("OK0") x: String, @Ann("OK1") y: String){}), "2")
+fun testClass(clazz: Class<*>, name: String, methodTest: (Method, String) -> Unit) {
+    val invokes = clazz.getDeclaredMethods().single() { !it.isBridge() }
+    methodTest(invokes, name)
+}
+
+@Ann("OK")
+fun annotatedFunction(@Ann("OK0") x: String, @Ann("OK1") y: String) {
+}
 
+fun box(): String {
+    testClass(Test.apply(@Ann("OK") fun(){}), "1", ::testMethod)
+    testClass(Test.applyABC(@Ann("OK") fun(@Ann("OK0") x: String, @Ann("OK1") y: String){}), "2", ::testMethod)
+    testClass(Test.applyABC(::annotatedFunction), "3", ::testMethodNoAnnotations)
```

Comment:
Maybe it's best to add this as a separate test instead?

Reply:
@madsager Please move to separate test

Reply:
Done, thanks!